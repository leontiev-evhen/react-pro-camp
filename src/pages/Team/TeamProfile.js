import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Figure } from 'react-bootstrap';
import './style.css';

function TeamProfile({ team }) {
	return (
		<div className="team-section">
			<Row>
				<Col md={3}>
					<Figure>
						<Figure.Image width={171} height={180} alt="logo" src={team.logo} />
						<Figure.Caption>
							<div>{team.venue_city}</div>
							<div>{team.venue_address}</div>
						</Figure.Caption>
					</Figure>
				</Col>
				<Col md={9}>
					<h1>{team.name}</h1>
					<ul>
						<li>
							<b>Country:</b> {team.country}
						</li>
						<li>
							<b>Founded:</b> {team.founded}
						</li>
						<li>
							<b>Stadium:</b> «{team.venue_name}»
						</li>
						<li>
							<b>Capacity:</b> {team.venue_capacity}
						</li>
					</ul>
				</Col>
			</Row>
		</div>
	);
}

TeamProfile.propTypes = {
	team: PropTypes.shape({
		logo: PropTypes.string,
		venue_city: PropTypes.string,
		venue_address: PropTypes.string,
		name: PropTypes.string,
		country: PropTypes.string,
		founded: PropTypes.number,
		venue_name: PropTypes.string,
		venue_capacity: PropTypes.number,
	}).isRequired,
};

export default TeamProfile;
