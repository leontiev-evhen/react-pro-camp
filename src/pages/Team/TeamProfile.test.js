import React from 'react';
import { shallow } from 'enzyme';
import TeamProfile from './TeamProfile';

describe('TeamProfile', () => {
	const props = {
		team: {
			logo: 'logo.png',
			venue_city: 'London',
			venue_address: 'Baker street',
			name: 'Arsenal',
			country: 'England',
			founded: 1989,
			venue_name: 'Stadium name',
			venue_capacity: 20000,
		},
	};

	const listOfValues = [
		'Country: England',
		'Founded: 1989',
		'Stadium: «Stadium name»',
		'Capacity: 20000',
	];

	const Component = shallow(<TeamProfile {...props} />);

	it('TeamProfile component render correctly', () => {
		expect(Component).toMatchSnapshot();
	});

	it('TeamProfile name', () => {
		expect(
			Component.find('.team-section')
				.find('h1')
				.text()
		).toEqual(props.team.name);
	});

	it('TeamProfile logo', () => {
		expect(
			Component.find('.team-section')
				.find('FigureImage')
				.prop('src')
		).toEqual(props.team.logo);
	});

	it('TeamProfile city', () => {
		expect(
			Component.find('.team-section')
				.find('FigureCaption')
				.find('div')
				.first()
				.text()
		).toEqual(props.team.venue_city);
	});

	it('TeamProfile address', () => {
		expect(
			Component.find('.team-section')
				.find('FigureCaption')
				.find('div')
				.last()
				.text()
		).toEqual(props.team.venue_address);
	});

	it('TeamProfile list of value', () => {
		Component.find('.team-section')
			.find('li')
			.forEach((node, key) => {
				expect(node.text()).toEqual(listOfValues[key]);
			});
	});
});
