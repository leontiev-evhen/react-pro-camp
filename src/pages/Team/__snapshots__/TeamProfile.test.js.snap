// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamProfile TeamProfile component render correctly 1`] = `
<div
  className="team-section"
>
  <Bootstrap(Row)>
    <Col
      md={3}
    >
      <Figure>
        <FigureImage
          alt="logo"
          fluid={true}
          height={180}
          src="logo.png"
          width={171}
        />
        <FigureCaption>
          <div>
            London
          </div>
          <div>
            Baker street
          </div>
        </FigureCaption>
      </Figure>
    </Col>
    <Col
      md={9}
    >
      <h1>
        Arsenal
      </h1>
      <ul>
        <li>
          <b>
            Country:
          </b>
           
          England
        </li>
        <li>
          <b>
            Founded:
          </b>
           
          1989
        </li>
        <li>
          <b>
            Stadium:
          </b>
           «
          Stadium name
          »
        </li>
        <li>
          <b>
            Capacity:
          </b>
           
          20000
        </li>
      </ul>
    </Col>
  </Bootstrap(Row)>
</div>
`;
